articles/rendering-html-pages-as-a-http-server-response-using-node-js/index.md

```diff
@@ -0,0 +1,333 @@
+
+Let us demonstrate how `__dirname` works with a  simple example that `console.log(__dirname)`.
+
+```js
+const express = require("express");
+const app = express();
+
+app.listen(3000, () => {
+  console.log("Application started and Listening on port 3000");
+});
+
+app.get("/", (req, res) => {
+  console.log(__dirname)
+});
+```
+
+Open the route `http://localhost:3000/` on the browser. This will print the values of `__dirname` into the console.
+
+![](console-log-dirname.jpg)
+
+As you can see, it gives you the exact path to reach your server location.
+
+Now we can send HTML files to the server using `__dirname`.
+
+To start with, I have created an HTML form (`index.html`) and included some CSS styling (`app.css`), as shown below.
+
+```html
+<!DOCTYPE html>
+<html lang="en"> 
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="app.css">
+  <link rel="stylesheet"
+  href="//netdna.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.no-icons.min.css">
+  <title>html form</title>
+</head>
+<body>
+<div class="subscribe-container">
+  <form method="POST">
+    <input type="email" name="youremail" placeholder="Email Address" required>
+    <input type="text" name="yourname" placeholder="Name" required>
+    <input class="subscribe-button" type="submit" value="Subscribe">
+  </form>
+</div>
+</body>
+</html>
+```
+
+`app.css`
+
+```css
+.subscribe-container {
+    max-width: 800px;
+    margin: 60px auto;
+    background-color: rgba(130, 184, 219, 0.5);
+    border: 5px solid rgb(98, 143, 228);
+}
+.subscribe-container form {
+    display: flex;
+    flex-wrap: wrap;
+}
+.subscribe-container form input {
+    margin: 15px;
+    border: 1px solid rgb(98, 143, 228);
+    padding: 0.4rem;
+}
+.subscribe-container form input {
+    flex: 1 1 200px;
+}
+
+.subscribe-container form input[type="submit"] {
+    border-radius: 3px;
+    background-color: rgba(17, 228, 10, 0.5);
+color: rgb(98, 143, 228);
+}
+
+.subscribe-container form input[type="email"] {
+    flex: 1 1 250px;
+}
+```
+
+Let us render the file into the server.
+
+```js
+const express = require("express");
+const app = express();
+
+app.listen(3000, () => {
+  console.log("Application started and Listening on port 3000");
+});
+
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+```
+
+![](html-page-as-server-response.jpg)
+
+However, this did not load the CSS styling. `app.css` is a static file. To [static server files](https://expressjs.com/en/starter/static-files.html) included in the `index.html` use `express.static` as shown in the example below;
```

Comment:
I think the second sentence is incomplete. Check that. It should be, to load static server files?